api/backups/restore.go

```diff
@@ -23,6 +28,15 @@ var (
 	}
 )
 
+// isUpgradeInProgressErr returns whether or not the error
+// is an "upgrade in progress" error.  This is necessary as
+// the error type returned from a facade call is rpc.RequestError
+// and we cannot use params.IsCodeUpgradeInProgress
+func isUpgradeInProgressErr(err error) bool {
```

Comment:
I think this is wrong. you probably want somethign like

switch err := err.(type) {
case *rpc.RequestError:
      return err.Code == params.CodeUpgradeInProgress
default:
      return false
}